atom/browser/api/lib/web-contents.coffee

```diff
@@ -34,6 +34,12 @@ module.exports.wrap = (webContents) ->
     do (name, method) ->
       webContents[name] = -> method.apply controller, arguments
 
+  # Service-worker utilities.
+  webContents.hasServiceWorker = (callback) ->
+    @_hasSW @getUrl(), callback
```

Comment:
The url returned by C++ `NavigationController::GetURL` should work for this case.